nvssclient/Controllers/RecordController.cs

```diff
@@ -251,36 +251,88 @@ public IActionResult AliasRecordHandler([FromBody] Dictionary<string, object> da
+                    // for each set of business ids, find the latest submitted message and its status
+                    List<MessageItem> latestMsgs = new List<MessageItem>();
+                    foreach (MessageItem ids in uniqueBussinessIds)
+                    {
+                        var message = context.MessageItems.Where(s => s.CertificateNumber == ids.CertificateNumber && s.DeathYear == ids.DeathYear && s.DeathJurisdictionID == ids.DeathJurisdictionID).OrderByDescending(s => s.CreatedDate).FirstOrDefault();
+                        latestMsgs.Add(message);
+                    }
+
+                    return latestMsgs;
+                }
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine("Error Retrieving record statuses: {0}", e);
+                return BadRequest();
+            }
+
+        }
+
         // GET: Record Status
-        // Retrieves the most recent MessageItem with business identifiers that match the provided parameters
+        // Retrieves all the messages sent with the specified business ids
         // deathYear: the year of death in the VRDR record
         // jurisditionId: the jurisdiction Id in the VRDR record
         // certNo: the 5 digit certificate number in the VRDR record
-        [HttpGet("status/{deathYear}/{jurisdictionId}/{certNo}")]
-        public ActionResult<RecordResponse> GetRecordStatus(uint deathYear, string jurisdictionId, string certNo)
+        [HttpGet("{deathYear}/{jurisdictionId}/{certNo}")]
+        public ActionResult<List<RecordResponse>> GetRecordStatus(uint deathYear, string jurisdictionId, string certNo)
         {
             try 
             {            
-                // Return the most recent message with the given business identifiers and it's status
+                // Return all messages with the given business identifiers and their responses if available
                 using (var scope = _scopeFactory.CreateScope()){
                     var context = scope.ServiceProvider.GetRequiredService<AppDbContext>();
                     uint certNoInt = UInt32.Parse(certNo);
-                    var message = context.MessageItems.Where(s => s.CertificateNumber == certNoInt && s.DeathYear == deathYear && s.DeathJurisdictionID == jurisdictionId).OrderByDescending(s => s.CreatedDate).FirstOrDefault();
-                    if (message == null) {
-                        Console.WriteLine("Error Retrieving status, no record was found for the provided identifiers.");
+                    List<MessageItem> messages = context.MessageItems.Where(s => s.CertificateNumber == certNoInt && s.DeathYear == deathYear && s.DeathJurisdictionID == jurisdictionId).OrderByDescending(s => s.CreatedDate).ToList();
+                    if (messages == null) {
+                        Console.WriteLine("Error, no messages were found for the provided identifiers.");
                         return NotFound("Record not found");
```

Comment:
I think this NotFound does make sense as is since it's a response for a particular identifier.